CMakeLists.txt

```diff
@@ -0,0 +1,17 @@
+cmake_minimum_required(VERSION 3.16)
+project(tetris CXX)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
```

Comment:
It's functionally identical in this case. I understand why compile features are the better choice for libraries that need to ensure a minimum standard is used by downstream targets. There are no downstream targets in this case though.

Reply:
I chose C++ 17 because I wasn't sure what Kofybrek used to build this on their machine. Looks like it will still compile in C++11 mode so we could potentially lower the standard requirement but I wanted to choose a more modern language standard to enable newer features. I don't suspect many people will struggle to find a C++17 compliant compiler these days.

Reply:
My comment wasn't really about the standard level itself, but the hardcoding of variables intended to be passed from the CLI/presets.

Well, the PR is merged already anyway. If you want to follow up with fixing this, then now you know how to do that.

Reply:
I got this pattern from CMake maintainer Crag Scott: https://crascit.com/2015/03/28/enabling-cxx11-in-cmake/

It makes total sense that in specifying the abstract build model for this project, which is the whole point of CMake, we also specify the language standard. Relying on users to know what flags to add is asking for trouble when inevitably many people do not do that. The expectation should be that a CMake project may be built with two commands:
```
cmake -S . -B build
cmake --build build
```

Reply:
Ensuring that a project builds without extensions is the developer's responsibility, not a requirement for an end-user and especially not a build requirement for the project. C++'s source level backwards compatibility ensures that declaring a minimal standard requirement, which is exactly what compile features do.

Hardcoding these variables is in fact, counterintuitively, declare an "at most" requirement. See the comment regarding decaying [here](https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html#prop_tgt:CXX_STANDARD).  
Furthermore, hardcoding these variables makes it impossible to test if a project builds with a certain standard version without having to patch the lists files. [Here](https://github.com/friendlyanon/daw_json_link/blob/cmake/.github/workflows/ci.yml#L65) is an example for testing multiple standards using the intended way.  
And if you made these variables cache variables instead, now you just made it impossible to vendor your project.

If you are curious what a well-structured CMake tooling looks like, please check [cmake-init](https://github.com/friendlyanon/cmake-init) out.